spec/puppet_blacksmith/forge_shared.rb

```diff
@@ -10,6 +10,7 @@
 RSpec.shared_context 'forge' do
   subject { Blacksmith::Forge.new(username, password, forge) }
 
+  let(:api_key) { 'e52f78b62e97cb8d8db6659a73aa522cca0f5c74d4714e0ed0bdd10000000000' }
```

Comment:
This is unused, so why add it?

You can also drop the duplicate `let(:module_name)`, since one will suffice. I think `let(:module_name) { 'test' }` should be dropped.

Reply:
if we drop the api_key mock we get:
```
1) Blacksmith::Forge push when using a Forge API key push the module
     Failure/Error: subject.api_key = api_key

     NameError:
       undefined local variable or method `api_key' for #<RSpec::ExampleGroups::BlacksmithForge_2::Push::WhenUsingAForgeAPIKey "push the module" (./spec/puppet_blacksmith/forge_spec.rb:81)>
     # ./spec/puppet_blacksmith/forge_spec.rb:82:in `block (4 levels) in <top (required)>'
     # ./vendor/bundle/ruby/3.0.0/gems/webmock-3.[18](https://github.com/voxpupuli/puppet-blacksmith/actions/runs/4919454151/jobs/8787088178?pr=119#step:5:19).1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

Reply:
Yes, I missed it was a shared context.